Stellar-ledger-entries.x

```diff
@@ -602,11 +641,27 @@ case CONTRACT_DATA:
     {
         Hash contractID;
         SCVal key;
+        ContractDataType type;
+
+        // ContractLedgerEntryType t is part of key type, but is inside a switch in the actual entry,
```

Comment:
I'm not sure I understand that, can't we use just the enum here and then use the xdr discriminant when building the key from the value?

Reply:
I think the issue is that for switches, `XDR` puts in some overhead, so the underlying struct for the `LedgerKey` union would be different than that of the `LedgerEntry` if we just had the enum without the shell switch. This may not be a big deal, but for every other type it looked like the `LedgerKey` was exactly the same as the beginning of the corresponding `LedgerEntry`, so I maintained that here but it might not be necessary.

Reply:
I'm not sure if there is any particular reason for these to be matching - not sure if that's a requirement and not a coincedence.